services/apps/alcs/src/providers/typeorm/migrations/1716848951501-update_day_calculations.ts

```diff
@@ -0,0 +1,116 @@
+                alcs.application_paused t2
+                WHERE
+                COALESCE(t1.end_date, NOW()) >= t2.start_date
+                AND COALESCE(t1.end_date, NOW()) < COALESCE(t2.end_date, NOW()) --- use coalesce since end_date may be null
+                AND t1.application_uuid = t2.application_uuid)
+            LEFT JOIN alcs.application ON application.uuid = s1.application_uuid
+            WHERE
+                NOT EXISTS (
+                SELECT
+                    *
+                FROM
+                alcs.application_paused s2
+                WHERE
+                    s1.start_date > s2.start_date
+                    AND s1.start_date <= COALESCE(s2.end_date, NOW())
+                    AND s1.application_uuid = s2.application_uuid)
+                AND s1.application_uuid = ANY (p_ids)
+                AND s1.start_date <= COALESCE(application.decision_date, NOW())
+                GROUP BY
+                s1.application_uuid, s1.start_date) pauses) pauses2
+    WHERE
+        application_uuid = ANY (p_ids)
+    GROUP BY
+        application_uuid;
+    $function$
+    ;
+    `);
+
+    await queryRunner.query(`
+        CREATE OR REPLACE FUNCTION alcs.calculate_noi_active_days(p_ids uuid[])
+          RETURNS TABLE(noi_uuid uuid, active_days integer, paused_days integer)
+            LANGUAGE sql
+            AS $function$
+          SELECT
+            "uuid",
+            CASE 
+              WHEN "date_received_all_items" IS NULL THEN
+                NULL
+              WHEN "has_open" IS TRUE
+                AND "meeting_count" > 0 THEN
+                NULL
+              WHEN "end_date" IS NOT NULL THEN
+                COALESCE("decision_date", NOW())::date - "end_date"::date
+              WHEN "decision_date" IS NOT NULL THEN
+                "decision_date"::date - "date_received_all_items"::date
+              WHEN "meeting_count" = 0 THEN
+                NOW()::date - "date_received_all_items"::date
+            END AS active_days,
+            CASE
+              WHEN "date_received_all_items" IS NULL THEN
+                NULL
+              WHEN "has_open" IS TRUE
+                AND "meeting_count" > 0 THEN
+                NOW()::date - "start_date"::date
+            END AS paused_days
+          FROM (
+            SELECT
+              "notice_of_intent".uuid,
+              MAX("decision_date") AS decision_date,
+              MAX("date_received_all_items") AS date_received_all_items,
+              max("end_date") AS end_date,
+              max("start_date") AS start_date,
+              bool_or("end_date" IS NULL) AS has_open,
+              count("notice_of_intent_meeting"."uuid") AS meeting_count
+            FROM
+              "alcs"."notice_of_intent"
+            LEFT JOIN "alcs"."notice_of_intent_meeting" ON 
+              "notice_of_intent"."uuid" = "notice_of_intent_meeting"."notice_of_intent_uuid"
+              AND start_date <= COALESCE("decision_date", NOW())
```

Comment:
This line was changed, adding AND start_date <= COALESCE("decision_date", NOW())